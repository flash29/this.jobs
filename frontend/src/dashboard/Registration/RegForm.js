import React, { Component } from 'react';
import { useState} from 'react';
import {Form, FormGroup, FormControl, Button } from 'react-bootstrap';
import { useNavigate } from "react-router-dom";
import {H} from 'react-headings';
import './Registration.css'



function RegForm(props) {

    const [postData, setPostData] = useState({ userName : '', userEmail : '', password : '' });
    let navigate = useNavigate();

    const handleRegisterClick = () =>{
        console.log(postData);
        fetch('/auth/register', {
            method : 'POST', 
            headers:{'Content-type':'application/json'},
            body:JSON.stringify(postData),
        }).then(response => response.json()).then(data => {
            console.log(data);
            setPostData({ userName: '', userEmail : '', password : ''});
            //window.location.reload(false)
            let path = "/";
            navigate(path);
        }).catch(error => console.log('error', error))
    }

    
    const handleLoginClick = () => {
        let path = "/";
        navigate(path);
    }

    return (
        <div className = "divStyle">
            <H className = "title"> this.jobs</H>
            <Form className="LoginForm" id="loginForm">
                <FormGroup controlId="formEmail" className = "FormComp w-50">
                    <FormControl placeholder="Enter your name" className = "inpBox" onChange={(e) => setPostData({...postData, userName : e.target.value}) }/>
                </FormGroup>
                <FormGroup className = "FormComp w-50">
                    <FormControl type="email" placeholder="Email Address" className = "inpBox" onChange={(e) => setPostData({...postData, userEmail : e.target.value}) }/>
                </FormGroup>
                <FormGroup controlId="formPassword" className = "FormComp w-50" onChange = {(e) => setPostData({...postData, password : e.target.value})}>
                    <FormControl type="password" className = "FormComp" placeholder="Password" className = "inpBox"/>
                </FormGroup>
                <FormGroup controlId="formSubmit" className = "FormComp">
                    <Button  className = "buttonStyle"  onClick={handleRegisterClick}>
                        Register
                    </Button>
                    {/* <Button  className = "buttonStyle"  onClick={handleLoginClick}>
                        Login
                    </Button> */}
                </FormGroup>
            </Form>
        </div>
    )
}


export default RegForm;