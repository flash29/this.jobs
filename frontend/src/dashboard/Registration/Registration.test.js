import React from 'react';
import ReactDom from 'react-dom';
import { Router, BrowserRouter } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import {createMemoryHistory} from 'history'
import RegPage from './Registration';
import RegForm from './RegForm';


import "@testing-library/jest-dom/extend-expect";
import { shallow, mount, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';

configure({adapter: new Adapter()});

test('Contains title', () => {
  render(
  <BrowserRouter>
    <RegPage />
  </BrowserRouter>);
  expect(screen.getByText(/this.jobs/i)).toBeInTheDocument();
});


test("Name input field working", () => {
    render(
        <BrowserRouter>
          <RegPage />
        </BrowserRouter>);
    expect(screen.queryByPlaceholderText(/Enter your name/i)).toBeInTheDocument();
});

test("Email input field working", () => {
    render(
        <BrowserRouter>
          <RegPage />
        </BrowserRouter>);
    expect(screen.queryByPlaceholderText(/Email Address/i)).toBeInTheDocument();
});

test("password input field working", () => {
    render(
        <BrowserRouter>
          <RegPage />
        </BrowserRouter>);
    expect(screen.queryByPlaceholderText(/Password/i)).toBeInTheDocument();
});

// test('Login Button test', () => {
//     const {debug, getByText} = render(<BrowserRouter>
//         <RegPage />
//       </BrowserRouter>);
//     expect(getByText('Login').tagName).toBe('BUTTON')
//     fireEvent.click(getByText('Login'))
//     //debug()
// });

test('Registration Button test', () => {
    const {debug, getByText} = render(<BrowserRouter>
        <RegPage />
      </BrowserRouter>);
    expect(getByText('Register').tagName).toBe('BUTTON')
    fireEvent.click(getByText('Register'))
    //debug()
});

test('render correctly the Registration Page component', () => {  
    const TextInputComponent = renderer.create(<BrowserRouter>
        <RegPage />
      </BrowserRouter>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

test('render correctly the Registration Form component', () => {  
    const TextInputComponent = renderer.create(<BrowserRouter>
        <RegForm />
      </BrowserRouter>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

test('rendering a component that users /auth/login', () => {
    const history = createMemoryHistory()
    const route = '/auth/login'
    history.push(route)
    render(
      <BrowserRouter history={history}>
        <RegForm />
      </BrowserRouter>)
});

test('Sending the post data to /auth/register', () => {
    const history = createMemoryHistory()
    const route = '/auth/register'
    history.push(route)
    render(
      <BrowserRouter history={history}>
        <RegForm />
      </BrowserRouter>)
});

test('Redirecting to login-page ', () => {
    const history = createMemoryHistory()
    const route = '/'
    history.push(route)
    render(
      <BrowserRouter history={history}>
        <RegForm />
      </BrowserRouter>)
});
