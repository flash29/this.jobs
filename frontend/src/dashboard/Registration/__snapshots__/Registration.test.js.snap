// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly the Registration Form component 1`] = `
<div
  className="divStyle"
>
  <h1
    className="title"
  >
     this.jobs 
  </h1>
  <h1 />
  <form
    className="LoginForm"
    id="loginForm"
  >
    <div
      className="FormComp w-50"
    >
      <input
        className="inpBox form-control"
        id="formEmail"
        onChange={[Function]}
        placeholder="Enter your name"
      />
    </div>
    <div
      className="FormComp w-50"
    >
      <input
        className="inpBox form-control"
        onChange={[Function]}
        placeholder="Email Address"
        type="email"
      />
    </div>
    <div
      className="FormComp w-50"
      onChange={[Function]}
    >
      <input
        className="inpBox form-control"
        id="formPassword"
        placeholder="Password"
        type="password"
      />
    </div>
    <div
      className="FormComp"
    >
      <button
        className="buttonStyle btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Register
      </button>
    </div>
  </form>
</div>
`;

exports[`render correctly the Registration Page component 1`] = `
<div
  className="Registration"
>
  <div
    className="divStyle"
  >
    <h1
      className="title"
    >
       this.jobs 
    </h1>
    <h1 />
    <form
      className="LoginForm"
      id="loginForm"
    >
      <div
        className="FormComp w-50"
      >
        <input
          className="inpBox form-control"
          id="formEmail"
          onChange={[Function]}
          placeholder="Enter your name"
        />
      </div>
      <div
        className="FormComp w-50"
      >
        <input
          className="inpBox form-control"
          onChange={[Function]}
          placeholder="Email Address"
          type="email"
        />
      </div>
      <div
        className="FormComp w-50"
        onChange={[Function]}
      >
        <input
          className="inpBox form-control"
          id="formPassword"
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="FormComp"
      >
        <button
          className="buttonStyle btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Register
        </button>
      </div>
    </form>
  </div>
</div>
`;
